tools/everest_basic_template.go

```diff
@@ -0,0 +1,94 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/operator-framework/operator-registry/alpha/declcfg"
+	"github.com/operator-framework/operator-registry/alpha/template/basic"
+	"gopkg.in/yaml.v3"
+	"os"
+)
+
+const versionPrefix = "everest-operator.v"
+
+type EverestBasicTemplate struct {
+	basic.BasicTemplate
+}
+
+func (t *EverestBasicTemplate) readFromFile(veneerFile string) error {
+	yamlFile, err := os.ReadFile(veneerFile)
+	if err != nil {
+		return err
+	}
+
+	var yamlData interface{}
+	err = yaml.Unmarshal(yamlFile, &yamlData)
+	if err != nil {
+		return err
+	}
+
+	jsonData, err := json.Marshal(yamlData)
+	if err != nil {
+		return err
+	}
+
+	return json.Unmarshal(jsonData, t)
```

Comment:
Is this JSON marshalling/unmarshalling really needed? Couldn't we directly unmarshal the yaml?

Reply:
no we can't, bc the original structure doesn't have the yaml tags and can't work with the blob. so we first need to go through the json and create the proper struct.